features/timeseries/src/main/java/org/opennms/netmgt/timeseries/integration/dao/TimeseriesResourceStorageDao.java

```diff
@@ -0,0 +1,331 @@
+
+    private static final Logger LOG = LoggerFactory.getLogger(TimeseriesResourceStorageDao.class);
+
+    @Autowired
+    private TimeSeriesStorage storage;
+
+    @Autowired
+    private Context m_context;
+
+    @Autowired
+    @Setter
+    private TimeseriesSearcher searcher;
+
+    @Autowired
+    private TimeseriesWriter writer;
+
+    @Autowired
+    private SearchableResourceMetadataCache searchableCache;
+
+    @Override
+    public boolean exists(ResourcePath path, int depth) {
+        Preconditions.checkArgument(depth >= 0, "depth must be non-negative");
+        if (hasCachedEntry(path, depth, depth)) {
+            return true; // cache hit!
+        }
+
+        return searchFor(path, depth, false).size() > 0;
+    }
+
+    @Override
+    public boolean existsWithin(final ResourcePath path, final int depth) {
+        Preconditions.checkArgument(depth >= 0, "depth must be non-negative");
+        if (hasCachedEntry(path, 0, depth)) {
+            return true; // cache hit!
+        }
+
+        // The indices are structured in such a way that we need specify the depth
+        // so here we need to iterate over all the possibilities. We could add
+        // additional indices to avoid this, but it's not worth the additional
+        // writes, since the specified depth should be relatively small.
+        return IntStream.rangeClosed(0, depth)
+            .anyMatch(i -> searchFor(path, i, false).size() > 0);
+    }
+
+    @Override
+    public Set<ResourcePath> children(ResourcePath path, int depth) {
+        Preconditions.checkArgument(depth >= 0, "depth must be non-negative");
+        Set<ResourcePath> matches = Sets.newTreeSet();
+
+        SearchResults results = searchFor(path, depth, false);
+        for (Result result : results) {
+            // Relativize the path
+            ResourcePath child = toChildResourcePath(path, result.getResource().getId());
+            if (child == null) {
+                // This shouldn't happen
+                LOG.warn("Encountered non-child resource {} when searching for {} with depth {}. Ignoring resource.",
+                        result.getResource(), path, depth);
+                continue;
+            }
+            matches.add(child);
+        }
+
+        return matches;
+    }
+
+    @Override
+    public boolean delete(ResourcePath path) {
+        final SearchResults results = searchFor(path, 0, true);
+
+        if (results.isEmpty()) {
+            return false;
+        }
+
+        for (final Result result : results) {
+            for(String metricName: result.getMetrics()) {
+                Metric metric = Metric.builder()
+                        .tag(CommonTagNames.resourceId, result.getResource().getId())
+                        .tag(CommonTagNames.name, metricName)
+                        .tag(Metric.MandatoryTag.mtype.name(), Metric.Mtype.gauge.name()) // TODO Patrick: discuss with Jesse: where do we get the type from?
```

Comment:
@j-white let's discuss on Tuesday

Reply:
With our current model, we only know the type when we write - we don't know it when it read or delete.